virtcontainers/kata_agent.go

```diff
@@ -843,6 +848,43 @@ func (k *kataAgent) handleBlockVolumes(c *Container) []*grpc.Storage {
 	return volumeStorages
 }
 
+func (k *kataAgent) handlePidNamespace(grpcSpec *grpc.Spec, sandbox *Sandbox) (bool, error) {
+	sharedPidNs := false
+
+	for i, ns := range grpcSpec.Linux.Namespaces {
+		if ns.Type == string(specs.PIDNamespace) {
+			if ns.Path != "" {
+				if sandbox.state.Pid > 0 {
+					pidNsPath := fmt.Sprintf("/proc/%d/ns/pid", sandbox.state.Pid)
+
+					//  Check if pid namespace path is the same as the sandbox
+					if ns.Path == pidNsPath {
+						sharedPidNs = true
+					} else {
+						// Check for symlinks
+						ln, err := filepath.EvalSymlinks(ns.Path)
+						if err != nil {
+							return sharedPidNs, err
```

Comment:
This is very deeply indented. I'd suggest reworking to invert the logi and continue if conditions are not `true`:

```
for i, ns := range grpcSpec.Linux.Namespaces {
    if ns.Type != string(specs.PIDNamespace) {
        continue
    }

    if ns.Path == "" {
        continue
    }

    ...

}
```

Reply:
done